webui/react/src/components/ColumnPickerMenu.test.tsx

```diff
@@ -0,0 +1,123 @@
+  });
+  it('should reset', async () => {
+    const { onVisibleColumnChange } = setup();
+    fireEvent.click(await screen.findByRole('button'));
+    const columnId = initialVisibleColumns[0];
+    const displayName = projectColumns.find((c) => c.column === columnId)?.displayName;
+    fireEvent.click(await screen.findByText(displayName ?? ''));
+    expect(onVisibleColumnChange).toHaveBeenCalledWith(
+      initialVisibleColumns.filter((c) => c !== columnId),
+    );
+    const resets = await screen.findAllByText('Reset');
+    fireEvent.click(resets[0]);
+    expect(onVisibleColumnChange).toHaveBeenCalledWith(initialVisibleColumns);
+  });
+  it('should switch tabs and display correct columns', async () => {
+    setup();
+    fireEvent.click(await screen.findByRole('button'));
+    const tabs = new Set<string>(Object.values(locationLabelMap));
+    const testTab = async (tabName: string) => {
+      fireEvent.click(await screen.findByText(tabName));
+      const locationForTab = _.findKey(locationLabelMap, (v) => v === tabName);
+      const columnsForLocation = projectColumns.filter((c) => c.location === locationForTab);
+      const column = columnsForLocation[0];
+      expect(await screen.findByText(column.displayName)).toBeInTheDocument();
+    };
+    Array.from(tabs).forEach((tabName: string) => {
+      testTab(tabName);
+    });
+  });
+  it('should show all', async () => {
+    const { onVisibleColumnChange } = setup();
+    fireEvent.click(await screen.findByRole('button'));
+    const showAlls = await screen.findAllByText('Show all');
+    fireEvent.click(showAlls[0]);
+    const locationColumns = projectColumns
+      .filter((c) => c.location === locations[0])
+      .map((c) => c.column);
+    const addedColumns = _.difference(locationColumns, initialVisibleColumns);
+    expect(onVisibleColumnChange).toHaveBeenCalledWith([...initialVisibleColumns, ...addedColumns]);
+  });
+  it('should hide all', async () => {
+    const locationColumns = projectColumns
+      .filter((c) => c.location === locations[0])
+      .map((c) => c.column);
+    const { onVisibleColumnChange } = setup(locationColumns);
+    fireEvent.click(await screen.findByRole('button'));
+    const hideAlls = await screen.findAllByText('Hide all');
+    fireEvent.click(hideAlls[0]);
+    expect(onVisibleColumnChange).toHaveBeenCalledWith([]);
+  });
+  it('should filter', async () => {
+    setup();
+    fireEvent.click(await screen.findByRole('button'));
+    const includedColumn = projectColumns.find((c) => c.column === initialVisibleColumns[0]);
+    const excludedColumn = projectColumns.find((c) => c.column === initialVisibleColumns[1]);
+    fireEvent.change(await screen.findByRole('textbox'), includedColumn?.displayName);
+    waitFor(async () => {
+      expect(await screen.findAllByText(includedColumn?.displayName ?? '')).toHaveLength(1);
+      expect(await screen.findAllByText(excludedColumn?.displayName ?? '')).toHaveLength(0);
+    });
```

Comment:
bug: looks like these test assertions are passing due to timeout. the test passes with any values such as `toHaveLength(1000);` i think its because of the `waitFor`.